### PR TITLE
Allow the namespace to be empty

### PR DESCRIPTION
**Pull request by cscheffler**
I would like to be able to support an empty namespace in getconf. An empty namespace will not get added to environment variable names.

This is a backwards compatible change since specifying a namespace will still get you the current behaviour. Setting the namespace no `None` will allow you to access `section.key` using the environment variable `SECTION_KEY`. It is not possible to achieve this at the moment, since setting the namespace to the empty string would result in an underscore at the beginning: `_SECTION_KEY`.

We have a use case for not wanting namespaces: Each of our services runs in a separate Docker container (so there is almost no risk of clashing environment variables) and the services need to know how to find each other. The host:port configuration (and other config, for example database) is made available in the environment variables, for example, `SERVICE1_HOST`, `SERVICE1_PORT`, `SERVICE2_HOST`, `SERVICE2_PORT`, but this should be the same on all containers.

Conceptually, this is the same as having a `.ini` file with one section per service. We just want to be able to access those sections and keys without being forced to put a namespace in front of everything. The Docker containers implicitly provide namespacing already.

### CODE REVIEW

File: tests/test_base.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def __exit__(self, *args, **kwargs):
 
 
 class ConfigGetterTestCase(unittest.TestCase):
-    example_directory = os.path.join(os.path.dirname(__file__), 'config/')
+    example_directory = os.path.abspath(os.path.join(os.path.dirname(__file__), 'config/'))
```
Comment (cscheffler) [inline tests/test_base.py]: This is an unrelated change. Many unit tests failed in my virtual environment because `os.path.dirname(__file__)` resolved to the relative path `tests/` and not to `/home/carl/code/getconf/tests/`. Using `abspath` fixes the problem.

**Comment by rbarrois** [comment]
This looks interesting. The initial design required a namespace to avoid acting on unrelated values (the environment can get quite polluted on an actual system), but your use-case is likely to occur more and more.

I'm still wondering about safeguarding against coding mistakes that would lead to an empty namespace when one was expected: since we're working with the external environment, it's difficult for a project to ensure that its program loads environment-related configuration properly, just as a typical app might not test its command-line interface.

In order to protect against those, I'd rather have users be explicit about "I don't want any namespace"; this could be forcing the use of a sentinel value:

```python
config = getconf.ConfigGetter(namespace=getconf.NO_NAMESPACE, [x, y, z])
```

What do you think?

**Comment by cscheffler** [comment]
That's a great idea.

Please see the updated PR. It now adds the `NO_NAMESPACE` constant to the module and explicitly checks `self.namespace` against that.

This also avoids the edge case, in the original commit, where a user might have set the namespace to the empty string, in which case environment variables would start with an underscore. The previous commit would have broken that behaviour, and therefore not have been backwards compatible.
